aws-sdk-core/lib/aws-sdk-core/cloudfront/signer.rb

```diff
@@ -0,0 +1,167 @@
+require 'base64'
+require 'uri'
+require 'time'
+require 'json'
+require 'openssl'
+
+module Aws
+  module CloudFront
+
+    # Allows you to create signed URLs for Amazon CloudFront resources
+    #
+    #     signer = Aws::CloudFront::UrlSigner.new
+    #     url = signer.signed_url(url,
+    #       key_pair_id: "cf-keypair-id",
+    #       private_key_path: "./cf_private_key.pem"
+    #     )
+    #
+    module Signer
+
+      # @option options [String] :key_pair_id
+      # @option options [String] :private_key
+      # @option options [String] :private_key_path
+      def initialize(options = {})
+        @key_pair_id = key_pair_id(options)
+        @private_key = private_key(options)
+      end
+
+      # create a signed Amazon CloudFront URL
+      # @param [String] url
+      # @option params [Time, DateTime, Date, String, Integer<timestamp>] :expires
+      # @option params [String<JSON>] :policy
+      def signed_url(url, params = {})
+        url_sections = url.split('://')
+        if url_sections.length < 2
+          raise ArgumentError, "Invaild URL:#{url}"
+        end
+        # removing wildcard character to get real scheme
+        scheme = url_sections[0].gsub('*', '')
+        uri = "#{scheme}://#{url_sections[1]}"
+        signed_content = signature(
+          resource: resource(scheme, uri),
+          expires: time(params[:expires]),
+          policy: params[:policy]
+        )
+
+        start_flag = URI.parse(uri).query ? '&' : '?'
+        uri = "#{uri}#{start_flag}#{signed_content.map{ |k, v| "#{k}=#{v}" }.join('&').gsub("\n", '')}"
```

Comment:
Perhaps separate `signed_content.map{ |k, v| "#{k}=#{v}" }.join('&').gsub("\n", '')` to a `signature` variable in another line to make this more readable?